envi/archs/amd64/__init__.py

```diff
@@ -144,3 +144,20 @@ def i_pextrd_q(self, op):
             self.i_pextrb(op, width=8)
         else:
             self.i_pextrb(op, width=4)
+
+    def i_idiv(self, op):
+        tsize = op.opers[0].tsize
+        if tsize == 8:
+            val = self.twoRegCompound(REG_RDX, REG_RAX, 8)
+            val = e_bits.signed(val, 16)
+            d = self.getOperValue(op, 0)
+            d = e_bits.signed(d, 8)
+            if d == 0:
+                raise envi.DivideByZero(self)
+            q = val // d
+            r = val % d
+
+            self.setRegister(REG_RAX, q)
+            self.setRegister(REG_RDX, r)
+        else:
+            e_i386.IntelEmulator.i_idiv(self, op)
```

Comment:
do we have unittests testing this functionality on amd64 for tsize==4 or other non-8?  
this can get tricky.

Reply:
in another emulator project, i wanted to demonstrate that the emulator did approximately as well as other well-tested emulators. it was easy to import unicorn and single step the emulators in sync, asserting that their CPU and memory state were the same at each step. i'd recommend the same here, rather than composing dozens of unit tests out of thin air.

Reply:
we've had that thought and discussion.  the question is how to automated it... and how best to *not* have our unittests dependent upon unicorn.  that means that we need some code to emulate, and that we need to perform any unicorn-testing step information in advance.  the question then is how to store the data statically, without filling up the hard drive.

i'm open to suggestions!  your idea is sound.  actually, i'm very open to PR's ;)

thanks @williballenthin .  this idea is a good one.